dadguide/monster_graph.py

```diff
@@ -79,30 +80,45 @@
   awakenings
   JOIN awoken_skills ON awakenings.awoken_skill_id = awoken_skills.awoken_skill_id"""
 
+EGG_QUERY = """SELECT
+   d_egg_machine_types.name AS type,
+   egg_machines.*
+FROM
+  egg_machines
+  JOIN d_egg_machine_types ON d_egg_machine_types.egg_machine_type_id = egg_machines.egg_machine_type_id
+"""
 
 class MonsterGraph(object):
     def __init__(self, database: DadguideDatabase):
         self.database = database
         self.graph = None
-        self.graph: nx.DiGraph
+        self.graph: networkx.MultiDiGraph
         self.edges = None
         self.nodes = None
         self.max_monster_id = -1
         self.build_graph()
 
     def build_graph(self):
-        self.graph = nx.MultiDiGraph()
+        self.graph = networkx.MultiDiGraph()
 
         ms = self.database.query_many(MONSTER_QUERY, (), DictWithAttrAccess, graph=self)
         es = self.database.query_many(EVOS_QUERY, (), DictWithAttrAccess, graph=self)
         aws = self.database.query_many(AWAKENINGS_QUERY, (), DictWithAttrAccess, graph=self)
+        ems = self.database.query_many(EGG_QUERY, (), DictWithAttrAccess, graph=self)
 
         mtoawo = defaultdict(list)
         for a in aws:
             awoken_skill_model = AwokenSkillModel(**a)
             awakening_model = AwakeningModel(awoken_skill_model=awoken_skill_model, **a)
             mtoawo[a.monster_id].append(awakening_model)
 
+        mtoegg = defaultdict(lambda: {'pem': False, 'rem': False})
+        for e in ems:
+            data = json.loads(e.contents)
+            type = 'pem' if e.type == "PEM" else 'rem'
+            for m in data:
+                mtoegg[int(m[1:-1])][type] = True
```

Comment:
can you leave a comment of why this works? `m[1:-1]` is pretty magical. maybe explain what the data looks like